packages/charts/src/utils/time_zone.ts

```diff
@@ -0,0 +1,32 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+const isValidTimeZone = (timeZone?: string): boolean => {
+  try {
+    Intl.DateTimeFormat(undefined, { timeZone });
+    return true;
+  } catch (error) {
```

Comment:
```suggestion
const isValidTimeZone = (timeZone?: string): timeZone is string => {
  if(!timeZone) {
    return false;
  }
  try {
    Intl.DateTimeFormat(undefined, { timeZone });
    return true;
  } catch (error) {
```

With that you can change the `getValidatedTimeZone` in:

```
export const getValidatedTimeZone = (specifiedTimeZone?: string): string =>
  isValidTimeZone(specifiedTimeZone) ? specifiedTimeZone : Intl.DateTimeFormat().resolvedOptions().timeZone;
```
Giving you a real validated timezone. The previous code will define an empty string as a valid timezone

Reply:
I added a test case to test a specified empty string 1d3db95bfbb98e021343cb6a6f0fa9a9842acc7e

Reply:
An empty string would make `Intl.DateTimeFormat(undefined, { timeZone })` throw (and the above empty string test case passes), but it was still useful to revisit `getValidatedTimeZone` as an undefined time zone does not throw, yet we want that to be also converted to local d6b21841a6587b000cd5e0d2ea465033bf2fc7f2